qiskit_experiments/framework/composite/parallel_experiment.py

```diff
@@ -54,20 +55,25 @@ def circuits(self):
 
         sub_circuits = []
         sub_qubits = []
+        sub_maps = []
         sub_size = []
         num_qubits = 0
 
         # Generate data for combination
-        for expr in self._experiments:
+        for sub_exp in self._experiments:
+
+            # Generate transpiled subcircuits
+            circuits = sub_exp._transpiled_circuits()
```

Comment:
`sub_exp` might be another composite experiment. Same handling is necessary as done in #667 . And allowing this situation makes it more complicated to count sub size. Probably we need to set some policy on construction of composite/nested composite experiment. To me parallel of parallel experiment doesn't make sense, for example. We should write some validator function. This should be covered by unittest.

Another complicated situation; does this hold?
```
assert BatchExperiment([ParallelExperiment(),...]) == ParallelExperiment([BatchExperiment(), ...])
```
in principle this should be satisfied unless there is no qubit overlap in component experiment.

Reply:
The current implementation is so that sub-experiment can be any BaseExperiment (with correct circuit and _transpiled_circuit methods), there is no special handling needed if sub experiment is another composite experiment. Nesting multiple parallel experiments is technically allowed, but would not be recommended from a performance perspective due to the added overhead of the extra transpilation for each level of nesting.

If I understood correctly the handling in 667 was specifically to avoid double transpilation of component circuits which is specific to batch experiment. Parallel experiment needs to do its final transpilation anyway after combining circuits so these special cases don't really matter here.

Reply:
If I understand correctly you should call `_transpiled_circuits` or `_batch_circuits` with `to_transpile=True`

Reply:
1. `_batch_circuits` with `to_transpile=True` is the same as `BatchExperiment._transpiled_circuits`.
2. I'm confused that all this happens inside `circuits`. I thought transpilation is relevant only in `_transpiled_circuits`. Can you clearly define the specification of `circuits`?

Reply:
ParallelExperiment's `circuits` method constructs the logical circuits on the relevant physical qubits like every BaseExperiment subclass is supposed to so it can use the default `BaseExperiment._transpiled_circuit` to handle final transpilation of these circuits to the correct layout.

How this is implemented inside the `circuits` method it calls each sub experiments `_transpiled_circuits` method to get those transpiled circuit, then remaps the relevant physical qubits in each component experiment to the corresponding logical qubits in the parallel circuit (same as before, but now with transpield circuits instead of component exps logical circuits).

Reply:
1. `T1.circuits` returns circuits acting on qubit 0, regardless of the physical qubit. What will `ParallelExperiment.circuits` return for a parallel experiment that combines a T1 on qubit 2 and a T1 on qubit 5?
2. Suppose that an experiment circuit contains gates that are not in the set of backend basis gates. Will these gates appear when calling `circuits` for that experiment? And will they appear when calling `ParallelExperiment.circuits` for a parallel experiment that combines two experiments of this type?

Reply:
@yaelbh 1. As I mentioned above it returns the logical circuits on qubits [0, 1] (with physical qubits set as [2, 5]), the same as it was before this PR.

For 2, if you call circuits with a backend set in the parallel experiment, the logical circuits will only contain bases gates for that backed because they will have been unrolled. If a backend wasn't set when you called this method they won't be unrolled. This may be different to the component experiment which generally won't do the unrolling in its circuit method.